specs/DevSpecs/telemetry.md

```diff
@@ -0,0 +1,187 @@
+ 
+## 1. Feature Dependencies 
+### 1.1 Dependencies on other features 
+Because telemetry will be baked into the `Input.Rating` control as a result of the implementation of this design, this feature is inherently dependent on the `Input.Rating` control. Externally, this feature is dependent on the functionality of the 1DS SDK. 
+ 
+### 1.2 Other features' dependencies on this feature 
+There are currently no other existing features that directly depend on this feature. However, future telemetry integration for other elements will be based on the design presented here. 
+ 
+## 2. Platform-Specific Considerations 
+UWP, JS, and .NET are all currently supported by 1DS, allowing for relatively seamless integration of telemetry for these platforms. Mobile platforms (ie., Android and iOS) are not fully supported by 1DS, so future communication and cooperation with the 1DS team to achieve support for these platforms is necessary. 
+ 
+## 3. Performance Implications 
+> “Perf-penalty in terms of runtime perf on foreground threads should not be affected; all our log event API calls are async. Background threads would kick in when there's time to upload. Total CPU time to upload would depend how much data you logged. In UTC mode - the perf penalty vs. trace logging would be near-zero. About the same.” - Max Golovanov (1DS) 
+
+We will look further into how 1DS affects startup time and memory usage with comparison tests. The real-world perf numbers would depend on telemetry volume per minute. The performance implications shouldn’t be major, but will exist.  
+ 
+ 
+## 4. Backwards Compatibility Concerns  
+If the renderer is not up to date, we will not be able to receive telemetry, which is not a problem because ‘Input.Rating’ would not work correctly without the up-to-date renderer.  
+ 
+## 5. Security Impact 
+Security should be considered in the situation of protecting the token and telemetry pipeline from malicious parties. Although Adaptive Cards is open source, we do not plan on exposing the telemetry token to the public. Thus, distribution of Adaptive Cards versions with telemetry should be limited to internal use cases only (within Microsoft Corp. only). 
+ 
+## 6. New APIs 
+This feature will introduce the IACLogger, which provides the following interface (see [section 9.1](#91-aclogger-telemetry-wrapper) for examples and implementation details):
+
+```
+interface IACLogger { 
+ 
+    logWarn(message: string): void; 
+ 
+    logErr(message: string): void; 
+ 
+    logVerbose(message: string): void; 
+ 
+    logInfo(message: string): void; 
+ 
+    logEvent(event: string, eventSourceName: string, correlationID: string, valueSet?: object): void; 
+ 
```

Comment:
We also exposing a way to specify/configure different telemetry providers right.. Should be up here too included as part of the interface